src/utils/recomposeBaseResolvers.js

```diff
@@ -0,0 +1,97 @@
+/* @flow */
```

Comment:
`recomposeBaseResolvers.js` is a not a good name.
Maybe `prepareResolversForDiscriminator.js` or something shorter. But from files tree, it should be clear what logic put inside.

Reply:
Or maybe better to create a `/discriminators` folder and put all files into it. So all logic and tests will be in one place.